services/github/github-size.service.js

```diff
@@ -1,21 +1,20 @@
 'use strict'
 
+const Joi = require('joi')
 const prettyBytes = require('pretty-bytes')
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const { makeLogo: getLogo } = require('../../lib/logos')
-const {
-  documentation,
-  checkErrorResponse: githubCheckErrorResponse,
-} = require('./github-helpers')
+const { nonNegativeInteger } = require('../validators')
+const { GithubAuthService } = require('./github-auth-service')
+const { documentation, errorMessagesFor } = require('./github-helpers')
+const { NotFound } = require('..')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class GithubSize extends LegacyService {
+const schema = Joi.alternatives(
+  Joi.object({
+    size: nonNegativeInteger,
+  }).required(),
+  Joi.array().required()
```

Comment:
This is so we can provide a more user friendly error message (`not a regular file`) vs. rejecting it at the schema level with the `invalid response data` error right?

Reply:
yep - exactly